ivy/functional/backends/jax/device.py

```diff
@@ -37,10 +38,14 @@ def dev(
     as_native: bool = False,
 ) -> Union[ivy.Device, jaxlib.xla_extension.Device]:
     if isinstance(x, jax.interpreters.partial_eval.DynamicJaxprTracer):
-        return ""
+        logging.warn(
```

Comment:
I think this should be an error right? If it fails silently and returns an empty string, that might be a bit hard to debug where we tried to access the device of `DynamicJaxprTracer`. Unless this is something native jax users expect, I think we should just raise an exception with the same message

Reply:
Oh okay makes sense, let's add a `ToDo` comment here then, sounds good!